src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -112,6 +112,13 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
         else:
             logger.debug("using haproxy bind address: %r", bind_addr)
 
+        # check if monitor needs to be bind on specific ip
+        monitoring_addr = None
+        if spec.monitoring_ip_addrs and spec.monitoring_ip_addrs.get(host) in self.mgr.cache.get_host_network_ips(host):
+            monitoring_addr = spec.monitoring_ip_addrs.get(host)
+        elif spec.monitoring_networks:
+            monitoring_addr = self.mgr.get_first_matching_network_ip(daemon_spec.host, spec, spec.monitoring_networks)
```

Comment:
if someone had bothered to set one or both of these fields and we still end up with `None` for the monitoring addr we should probably log something about it, since it's likely a mistake and they expect it to have picked a specific ip for the monitoring